server/src/main/java/org/opensearch/telemetry/tracing/TraceableRunnable.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+/**
+ * Wraps the runnable and add instrumentation to trace the {@link Runnable}
+ */
+public class TraceableRunnable implements Runnable {
```

Comment:
I think we could move this to libs/telemetry, pretty useful class not only in server context